server/mailet/integration-testing/src/test/java/org/apache/james/mailets/DKIMIntegrationTest.java

```diff
@@ -0,0 +1,163 @@
+    private static final MailetConfiguration DKIMSIGN_MAILET = MailetConfiguration.builder()
+        .matcher(All.class)
+        .mailet(DKIMSign.class)
+        .addProperty(
+            "signatureTemplate",
+            "v=1; s=selector; d=example.com; h=from:to:received:received; a=rsa-sha256; bh=; b=;")
+        .addProperty("privateKey", TESTING_PEM)
+        .build();
+
+    private static final MailetConfiguration DKIMVERIFY_MAILET = MailetConfiguration.builder()
+        .matcher(All.class)
+        .mailet(DKIMVerify.class)
+        .build();
+
+
+    private static final MailetConfiguration STUB_MAILET = MailetConfiguration.builder()
+        .matcher(All.class)
+        .mailet(ExtractAttributeStub.class)
+        .addProperty("attributeName", DKIMVerify.DKIM_AUTH_RESULT.asString())
+        .build();
+
+    private static final PublicKeyRecordRetriever MOCK_PUBLIC_KEY_RECORD_RETRIEVER = new MockPublicKeyRecordRetriever(
+            "v=DKIM1; k=rsa; p=MIGfMA0GCSqGSIb3DQEBAQUAA4GNADCBiQKBgQDYDaYKXzwVYwqWbLhmuJ66aTAN8wmDR+rfHE8HfnkSOax0oIoTM5zquZrTLo30870YMfYzxwfB6j/Nz3QdwrUD/t0YMYJiUKyWJnCKfZXHJBJ+yfRHr7oW+UW3cVo9CG2bBfIxsInwYe175g9UjyntJpWueqdEIo1c2bhv9Mp66QIDAQAB;",
+            "selector", "example.com");
+
+    @Rule
+    public TemporaryFolder temporaryFolder = new TemporaryFolder();
+    @Rule
+    public IMAPMessageReader imapMessageReader = new IMAPMessageReader();
+    @Rule
+    public SMTPMessageSender messageSender = new SMTPMessageSender(DEFAULT_DOMAIN);
+
+    private TemporaryJamesServer jamesServer;
+    private DataProbe dataProbe;
+    private List<Optional<String>> dkimAuthResults;
+
+    @Before
+    public void setup() throws Exception {
+        dkimAuthResults = new ArrayList<>();
+        ExtractAttributeStub.setDkimAuthResultInspector(value -> dkimAuthResults.add(value.map(result -> (String) result)));
+        MailetContainer.Builder mailetContainer = TemporaryJamesServer.SIMPLE_MAILET_CONTAINER_CONFIGURATION
+            .putProcessor(ProcessorConfiguration.transport()
+                .addMailet(DKIMSIGN_MAILET)
+                .addMailet(DKIMVERIFY_MAILET)
+                .addMailet(STUB_MAILET)
+                .addMailetsFrom(CommonProcessors.transport()));
+
+        jamesServer = TemporaryJamesServer
+            .builder()
+            .withBase(Modules.combine(
+                MemoryJamesServerMain.IN_MEMORY_SERVER_AGGREGATE_MODULE,
+                binder -> binder.bind(PublicKeyRecordRetriever.class).toInstance(MOCK_PUBLIC_KEY_RECORD_RETRIEVER)))
+            .withMailetContainer(mailetContainer)
+            .build(temporaryFolder);
+
+        dataProbe = jamesServer.getProbe(DataProbeImpl.class);
+        dataProbe.addDomain(DEFAULT_DOMAIN);
+
+        dataProbe.addUser(RECIPIENT, PASSWORD);
+    }
+
+    @After
+    public void tearDown() {
+        jamesServer.shutdown();
+    }
+
+    @Test
+    public void incomingMessageShouldBeReceivedSigned() throws Exception {
```

Comment:
the test name should explain that you want the DKIM verifier to check the signature.